src/main/java/io/cdap/plugin/sap/odata/odata4/OData4Client.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.sap.odata.odata4;
+
+import io.cdap.plugin.sap.odata.EntityType;
+import io.cdap.plugin.sap.odata.ODataClient;
+import io.cdap.plugin.sap.odata.ODataEntity;
+import io.cdap.plugin.sap.odata.PropertyMetadata;
+import org.apache.olingo.client.api.communication.request.retrieve.EdmMetadataRequest;
+import org.apache.olingo.client.api.communication.request.retrieve.ODataEntitySetIteratorRequest;
+import org.apache.olingo.client.api.communication.response.ODataRetrieveResponse;
+import org.apache.olingo.client.api.domain.ClientEntity;
+import org.apache.olingo.client.api.domain.ClientEntitySet;
+import org.apache.olingo.client.api.domain.ClientEntitySetIterator;
+import org.apache.olingo.client.core.ODataClientFactory;
+import org.apache.olingo.client.core.http.BasicAuthHttpClientFactory;
+import org.apache.olingo.commons.api.edm.Edm;
+import org.apache.olingo.commons.api.edm.EdmEntityType;
+import org.apache.olingo.commons.api.edm.EdmProperty;
+
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import javax.annotation.Nullable;
+import javax.ws.rs.core.MediaType;
+
+/**
+ * Provides handy methods to consume OData v4 services using Apache Olingo v4 as Client Library.
+ */
+public class OData4Client extends ODataClient {
+
+  private org.apache.olingo.client.api.ODataClient client;
+
+  /**
+   * @param rootUrl  URL of the OData service. The URL must end with an external service name
+   *                 (e.g., http://eccsvrname:8000/sap/opu/odata/sap/zgw100_dd02l_so_srv/).
+   * @param username username for basic authentication.
+   * @param password password for basic authentication.
+   */
+  public OData4Client(String rootUrl, String username, String password) {
+    super(rootUrl, username, password);
+    this.client = ODataClientFactory.getClient();
```

Comment:
does this client need to be closed at some point?

Reply:
The client does not have methods to close it. There is no need to close it as per [sample client code from Olingo documentation](https://olingo.apache.org/doc/odata4/tutorials/od4_basic_client_read.html#sample).